src/xrootiface.py

```diff
@@ -31,22 +34,34 @@ def _getxrdfor(endpoint):
     xrdfs[endpoint] = XrdClient.FileSystem(endpoint)
     return xrdfs[endpoint]
 
+
 def _geturlfor(endpoint):
   '''Look up the URL for a given endpoint: "default" corresponds to the defaultstorage one'''
   if endpoint == 'default':
     return defaultstorage
   return endpoint
 
-def _eosargs(ruid, rgid, atomicwrite=0, bookingsize=0):
-  '''One-liner to generate extra EOS-specific arguments for the xroot URL'''
-  return '?eos.ruid=' + ruid + '&eos.rgid=' + rgid + ('&eos.atomic=1' if atomicwrite else '') + \
-          (('&eos.bookingsize='+str(bookingsize)) if bookingsize else '') + '&eos.app=wopi'
 
-def _xrootcmd(endpoint, cmd, subcmd, ruid, rgid, args):
-  '''Perform the <cmd>/<subcmd> action on the special /proc/user path on behalf of the given uid,gid.
+def _eosargs(userid, atomicwrite=0, bookingsize=0):
+  '''Split userid into uid,gid and generate extra EOS-specific arguments for the xroot URL'''
+  try:
+    # try to assert that userid must follow a '%d:%d' format
```

Comment:
Maybe we can add an example of an user id in the method comment?

Reply:
Yes, what about '''Assume userid is in the form `uid:gid` and ...''' ?